mod_geoip.c

```diff
@@ -447,12 +453,12 @@ static int geoip_header_parser(request_rec * r)
 
             if (cfg->proxyHeaderMode ==
                 GEOIP_PROXY_HEADER_MODE_FIRST_NON_PRIVATE_IP) {
-                ipaddr = free_me = _get_ip_from_xff(ipaddr_ptr);
+                ipaddr = _get_ip_from_xff(r, ipaddr_ptr);
                 if (!ipaddr) {
                     ipaddr = _get_client_ip(r);
                 }
             } else {
-                ipaddr = free_me = (char *)calloc(8 * 4 + 7 + 1, sizeof(char));
+                ipaddr = (char *) apr_pcalloc(r->pool, 8 * 4 + 7 + 1);
```

Comment:
I agree that using the pool is much better than doing our own memory management.